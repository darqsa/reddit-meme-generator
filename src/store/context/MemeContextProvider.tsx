import { useReducer } from "react";
import memesReducer from "../reducers/memesReducer";
import MemeContext from "./MemeContext";

interface MemeContextProviderProps {
  children: JSX.Element | JSX.Element[];
}

const MemeContextProvider = ({
  children,
}: MemeContextProviderProps): JSX.Element => {
  const [memes, dispatch] = useReducer(memesReducer, []);
  return (
    <MemeContext.Provider value={{ memes, dispatch }}>
      {children}
    </MemeContext.Provider>
  );
};

export default MemeContextProvider;
