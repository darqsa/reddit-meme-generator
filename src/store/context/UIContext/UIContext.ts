import { createContext } from "react";
import { IUIContext } from "../../../interfaces/interfaces";

const UIContext = createContext<IUIContext>({
  ui: {
    isLoading: false,
    type: "loading",
  },
  dispatch: () => {},
});

export default UIContext;
