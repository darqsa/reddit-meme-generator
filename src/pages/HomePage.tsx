import { useContext } from "react";
import CardList from "../components/CardList/CardList";
import Header from "../components/Header/Header";
import Loading from "../components/Loading/Loading";
import Modal from "../components/Modal/Modal";
import UIContext from "../store/context/UIContext/UIContext";

const HomePage = (): JSX.Element => {
  const { feedback, isLoading } = useContext(UIContext);

  return (
    <>
      {feedback.isOpen && feedback.modalType === "error" ? (
        <Modal type="error" />
      ) : (
        <></>
      )}
      {feedback.isOpen && feedback.modalType === "succes" ? (
        <Modal type="success" />
      ) : (
        <></>
      )}
      {isLoading ? <Loading /> : <></>}
      <Header currentPage="mainPage" />
      <CardList currentPage="mainPage" />
    </>
  );
};

export default HomePage;
