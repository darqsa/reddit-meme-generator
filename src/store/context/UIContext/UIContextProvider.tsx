import { useReducer } from "react";
import { UIState } from "../../../interfaces/interfaces";
import UIReducer from "../../reducers/UIReducer/UIReducer";
import UIContext from "./UIContext";

interface UIContextProviderProps {
  children: JSX.Element | JSX.Element[];
}
const UIContextProvider = ({
  children,
}: UIContextProviderProps): JSX.Element => {
  const InitialUI: UIState = {
    isLoading: false,
    feedback: "succes",
  };

  const [{ feedback, isLoading }, dispatch] = useReducer(UIReducer, InitialUI);

  return (
    <UIContext.Provider value={{ dispatch, feedback, isLoading }}>
      {children}
    </UIContext.Provider>
  );
};

export default UIContextProvider;
