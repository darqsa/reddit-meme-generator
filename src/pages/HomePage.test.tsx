import { BrowserRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import MemeContextProvider from "../store/context/MemeContext/MemeContextProvider";
import HomePage from "./HomePage";
import UIContext from "../store/context/UIContext/UIContext";

describe("Given an HomePage page", () => {
  describe("When its rendered on screen", () => {
    test("Then it should show the home page", () => {
      const expectedHomePage = TestRenderer.create(
        <MemeContextProvider>
          <BrowserRouter>
            <HomePage />
          </BrowserRouter>
        </MemeContextProvider>
      );
      expect(expectedHomePage).toMatchSnapshot();
    });
  });

  describe("And UI context have the feedback Open and a succes message", () => {
    test("Then it should show the modal succes on screen", () => {
      const expectedHomePage = TestRenderer.create(
        <UIContext.Provider
          value={{
            feedback: { isOpen: true, modalType: "succes" },
            isLoading: false,
            dispatch: () => {},
          }}
        >
          <BrowserRouter>
            <HomePage />
          </BrowserRouter>
        </UIContext.Provider>
      );

      expect(expectedHomePage).toMatchSnapshot();
    });
  });

  describe("And UI context have feedback Open and an error message", () => {
    test("Then it should show the modal succes on screen", () => {
      const expectedHomePage = TestRenderer.create(
        <UIContext.Provider
          value={{
            feedback: { isOpen: true, modalType: "error" },
            isLoading: false,
            dispatch: () => {},
          }}
        >
          <BrowserRouter>
            <HomePage />
          </BrowserRouter>
        </UIContext.Provider>
      );

      expect(expectedHomePage).toMatchSnapshot();
    });
  });
});
