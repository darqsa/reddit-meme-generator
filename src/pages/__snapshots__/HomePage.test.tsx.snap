// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an HomePage page When its rendered on screen Then it should show the home page 1`] = `
Array [
  <div
    className="header-container"
  >
    <header
      className="sc-hKMtZM jvqPw header-container__title-container"
    >
      <img
        alt="Awesome meme generator logo, it's a frog"
        className="sc-eCYdqJ eoKxeA header-container__title-image"
        src="img/logo.png"
      />
      <h1
        className="sc-jSMfEi eIWBdb header-container__title"
      >
        AWESOME 
        <span
          className="sc-papXJ eRKGXE"
        >
          MEME 
        </span>
        GENERATOR
      </h1>
    </header>
    <section
      className="sc-gKXOVf iqGkus header-container__section-container"
    >
      <div
        className="sc-iBkjds eITUJN header-container__texts-container"
      >
        <p
          className="sc-ftvSup jccDpC header-container__text"
        >
          Click the “Generate memes” button to get 10 random memes, if you like a meme you can click the
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star header-container__text-icon"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          button to move it into your favourite pages.
        </p>
        <p
          className="sc-ftvSup jccDpC header-container__text"
        >
          In your favourite pages you can unlike a meme, edit it and also create your own meme.
        </p>
      </div>
      <div
        className="sc-jqUVSM jOaidU header-container__buttons-container"
      >
        <button
          className="sc-bczRLJ hwyGhm"
          onClick={[Function]}
        >
          Generate Memes
        </button>
        <a
          href="/favorites"
          onClick={[Function]}
        >
          <button
            className="sc-bczRLJ hwyGhm"
            onClick={[Function]}
          >
            My Favorites
          </button>
        </a>
      </div>
    </section>
  </div>,
  <div
    className="sc-dkzDqf hOjkB"
  >
    <ul
      className="meme-list"
    />
  </div>,
]
`;
